src/utilities/api-requests.ts

```diff
@@ -10,13 +15,23 @@ import axios from "axios";
  */
 export async function request<T>(uri: string, defaultValue: T): Promise<T> {
     let data = defaultValue;
-    await axios
-        .get(uri)
-        .then((res) => {
-            data = res.data as T;
-        })
-        .catch((error) => {
-            console.error(`Request to ${uri} failed with\n${error}`);
-        });
+    let attempt = 0;
+
+    while (data === defaultValue && attempt < 5) {
+        const response = await axios
+            .get<T>(uri)
+            .catch((reason: AxiosError) => handleError(uri, reason));
+
+        if (response) {
+            if (response.status === 200) {
```

Comment:
Is 200 OK actually the only Success HTTP-code we can get here? I suppose it is defined in the API. Not very resistant to change, however, if the API is updated to include more success codes.

Reply:
Fixed. It now accepts and HTTP codes in the range [200, 300).